src/main.rs

```diff
@@ -116,18 +116,21 @@ fn main() {
     });
 
     // spawn thread to maintain histogram snapshots
-    control_runtime.spawn(async {
-        while RUNNING.load(Ordering::Relaxed) {
-            // acquire a lock and update the snapshots
-            {
-                let mut snapshots = METRICS_SNAPSHOT.write().await;
-                snapshots.update();
+    {
```

Comment:
Curious why you chose to make this a separate scope/code block.

Reply:
Probably wasn't necessary, but keeps the `interval` locally scoped.